### PR TITLE
Changed default value of RowGroupColumnName from null to GroupId

### PR DESCRIPTION
**Pull request by michaelgsharp**
This is the final pr to fix issue #4749, changes the default value of RowGroupColumnName from null to GroupId.

### CODE REVIEW

**Comment by michaelgsharp** [comment]
@justinormont This is the final pr needed to close the changing default issue. Per our discussion, I have updated the test that was missing the GroupId column to include that column.

File: src/Microsoft.ML.FastTree/FastTreeArguments.cs [hunk 1]
```diff
@@ -301,6 +301,7 @@ public EarlyStoppingRankingMetric EarlyStoppingMetric
             public Options()
             {
                 EarlyStoppingMetric = EarlyStoppingRankingMetric.NdcgAt1; // Use L1 by default.
+                RowGroupColumnName = "GroupId"; // Use GroupId as default for ranking options.
```
Comment (antoniovs1029) [inline src/Microsoft.ML.FastTree/FastTreeArguments.cs:304]: **Suggestion**: Use `DefaultColumnNames.GroupId` instead, also the one in LightGbm

**Comment by frank-dong-ms** [comment]
Could you please share what's the impact of this change? Does it lead to faster training or accurate result or anything else?

**Comment by michaelgsharp** [comment]
Hey Frank, issue #4749 talks more about it, but basically we want to have good default values so that the default experience for end  users is good. If you do tree training without a group, it has bad results. By making the default use a group, then the default will be better results. In fact, only 1 of our tests didn't have a group, the rest did have a group for that reason.

File: src/Microsoft.ML.FastTree/FastTreeArguments.cs (hunk 1 above)
Comment (michaelgsharp) [reply src/Microsoft.ML.FastTree/FastTreeArguments.cs:304]: Good point, I will change that.

File: test/BaselineOutput/Common/EntryPoints/core_manifest.json [hunk 2]
```diff
@@ -7675,7 +7675,7 @@
           "Required": false,
           "SortOrder": 5.0,
           "IsNullable": false,
-          "Default": null
```
Comment (antoniovs1029) [inline test/BaselineOutput/Common/EntryPoints/core_manifest.json:7678]: **Question:** Just curious, did you have to manually change this? Or was it automatically updated simply by adding the constructor logic you added?

**Pull request merged**